client/httplib/httplib.go

```diff
@@ -654,3 +634,12 @@ func TimeoutDialer(cTimeout time.Duration, rwTimeout time.Duration) func(net, ad
 		return conn, err
 	}
 }
+
+func logFilter(next Filter) Filter {
+	return func(ctx context.Context, req *BeegoHTTPRequest) (*http.Response, error) {
+		log.Println(req)
+		resp, err := next(ctx, req)
+		log.Println(resp)
+		return resp, err
+	}
+}
```

Comment:
Please move this method to a new file.
We don't print `req` and `resp`, and we could use `httputil.DumpRequest` and `httputil.DumpResponse`.

The keypint is that we should check http header content-type, and print the body if we got expected content-type.

For example, if the content-type is application/json, we print body, or we don't print body.

Let's assume that you have a builder and it has two options:
```go
type logFilterBuilder struct{
    PrintableContentType []string // default value are those textual content-type
    MaxBodyLength  int // default 0
}
```
if we the content-type of request(or response) is in `PrintableContentType` array, we print the body.

When we print the body, if `MaxBodyLength <=0` we print whole body, or we print substring of body.

If you are still confused, and prefer online discussion, we could have a short discussion on zoom.

Reply:
@flycash There are three questiones need your help.
1. What is the relationship between `ShowDebug` and `logFilterBuilder`. As you said, we provide a default filter like Opentracing Filter, user can initialize it by themselves, and add it to `FilterChain`. So `ShowDebug` plays a what kind of  role?
2. User can customize the `PrintableContentType` parameter, do we need to verify whether the parameter really printable? (eg. The default value contains all kinds of printable content types, the user input parameter should be a subset of default.)
3. `logFilter` should print the request processed by `FilterChain` including `doRequest()` and response before filterChain processing?
I am willing to discuss online, when do you have time?

Reply:
1. I prefer that making `logFilter` as default filter which means that we will register this filter when httplib module initialized. So if `ShowDebug` is true, we log request and response, or `logFilter` do nothing. Another option is that we remove `ShowDebug` field, but user should register this filter by themselves.
2. I think users should be responsible for their choices. We only provide default value of `PrintableContentType`, but don't need to verify it when users change it.
3. `logFilter` should log request before `doRequest` and log response after `doReuest`. In our design, `doRequest` is the last "filter", it looks like: `logFilter.next.next...=doRequest`:
```go
func FilterChain(next) Filter{
    return func() {
     logRequest()
     next() // finally we will run doRequest.
     logResponse()
   }
}
```

I always have time except UTC+8 00:00 - 10:00. Please send me an email to tell me your time.